pkg/ccl/changefeedccl/sink_webhook.go

```diff
@@ -220,15 +267,88 @@ func (s *webhookSink) sendMessage(ctx context.Context, reqBody []byte) error {
 	return nil
 }
 
+// workerIndex assigns rows each to a worker goroutine based on the hash of its
+// primary key. This is to ensure that each message with the same key gets
+// deterministically assigned to the same worker. Since we have a channel per
+// worker, we can ensure per-worker ordering and therefore guarantee per-key
+// ordering.
+func (s *webhookSink) workerIndex(key []byte) uint32 {
+	return crc32.ChecksumIEEE(key) % uint32(s.parallelism)
+}
+
+// TODO (ryan min): add memory monitoring for inflight messages
+// inflightTracker wraps logic for counting number of inflight messages to
+// track when flushing sink, with error handling functionality. Implemented
+// as a wrapper for WaitGroup under a lock to block additional messages while
+// flushing.
+type inflightTracker struct {
+	// two mutexes are needed here since different goroutines will be accessing
+	// both the waitgroup and the flushErr at the same time. (i.e. when waiting,
+	// before Done() is called, the error is set, causes deadlock if the same
+	// lock is used)
+	wgMu struct {
```

Comment:
I think that putting the mutex and the waitgroup in this same structure is a bit confusing.  We should not need a mutex around a waitgroup since the waitgroup is already a concurrency safe datastructure.  However, I believe that this mutex is to prevent Add() calls while we are waiting on Flush()?  If that is correct, let's make the naming of this mutex and the comment reflect that.